[gh project item-edit] Fix number type

iamazeem opened the pull request:
Fixes #10342.

```shell
# number (float32)
$ gh project item-edit --project-id $projectId --id $itemId --field-id $fieldId --number 69.84
GraphQL: Number values cannot exceed 8 decimal places (updateProjectV2ItemFieldValue)

# number (float64)
$ bin/gh project item-edit --project-id $projectId --id $itemId --field-id $fieldId --number 69.84
Edited item "Test"
```

iamazeem committed:
[gh project item-edit] Fix number type

andyfeller commented:
@iamazeem : I'm catching up on #10342 to wrap up triaging how big of a problem we have beyond `gh project item-edit` such as other `gh project` commands we might have problems with.  

Generally, we prefer maintainers capturing acceptance criteria and marking issues as `help wanted` as a way of signaling they are available for contributions or `core` being work that maintainers want to do themselves.

While I wrap this up, what are you thoughts on how the associated tests should be updated to ensure this continues to work as expected?

https://github.com/cli/cli/blob/64b63f61f667639167e89082541f23c11fca5061/pkg/cmd/project/item-edit/item_edit_test.go#L14-L159

iamazeem committed:
Update tests

iamazeem commented:
@andyfeller

> I'm catching up on #10342 to wrap up triaging how big of a problem we have beyond `gh project item-edit` such as other `gh project` commands we might have problems with.
>
> Generally, we prefer maintainers capturing acceptance criteria and marking issues as `help wanted` as a way of signaling they are available for contributions or `core` being work that maintainers want to do themselves.

Please take your time.
I was able to verify and test it locally.
Wanted to verify it through CI as well.

> what are you thoughts on how the associated tests should be updated to ensure this continues to work as expected?

Looked into it and pushed https://github.com/cli/cli/pull/10374/commits/277f27442ce678d578a04b3feab72aa5265a78a6 where:
- added a new test with floating number for the CLI args, and
- updated GQL API number test for the floating point number.

Looks like the GQL API test may further be enhanced with the complete response. :thinking:
But, the other tests for the field values are not checking for that so I'm not so sure about that.
Please share how you think to test it against these cases. Thanks!

andyfeller reviewed (approved):
Looks good!

Working with @BagToad, we've removed the AC for client-side validation of `--number` max values in favor for server-side validation improvements with the Projects team.
